frontend/src/PastEncounterList.tsx

```diff
@@ -0,0 +1,196 @@
+    top: 20,
+    width: 1,
+  },
+  viewEncounterDialog: {
+    margin: theme.spacing(1)
+  }
+}));
+
+interface PastEncounterListProps {
+  user: firebase.User;
+  refresh: boolean; // Used to force refresh from server.
+}
+
+const headers = [
+  {
+    id: "time",
+    numeric: false,
+    disablePadding: false,
+    label: "Scheduled Time",
+  },
+  { id: "name", numeric: false, disablePadding: false, label: "Patient" },
+  { id: "title", numeric: false, disablePadding: false, label: "Title" },
+  {
+    id: "advocates",
+    numeric: false,
+    disablePadding: false,
+    label: "Advocates",
+  },
+  { id: "status", numeric: false, disablePadding: false, label: "Status" },
+  { id: "actions", numeric: false, disablePadding: false, label: "" },
+];
+
+export default function PastEncounterList({ user, refresh }: PastEncounterListProps) {
+  const classes = useStyles();
+  console.log("PastEncounterList");
+  const [encounters, setEncounters] = React.useState<Array<Encounter>>([])
+  const [viewOpen, setViewOpen] = React.useState(false);
+  const [viewEncounterIndex, setViewEncounterIndex] = React.useState<number>(0);
+  const [transcript, setTranscript] = React.useState(String);
+
+  const refreshEncounters = React.useCallback(() => {
+    console.log("Past Encounter List: refreshEncounters");
+    let listEncounters = firebase.functions().httpsCallable("listEncountersByStatus");
+    listEncounters({ userId: "myuser", status: EncounterState.COMPLETE })
+      .then((response) => {
+        let newEncounters = response.data.map((entry: Encounter) => {
+          console.log(JSON.stringify(entry));
+          return entry;
+        });
+        setEncounters(newEncounters);
+      })
+      .catch((err) => {
+        console.log("ERROR: " + JSON.stringify(err));
+      });
+  }, [setEncounters]);
+
+  const fetchTranscript = React.useCallback((encounterId: String) => {
+    console.log("Getting transcript");
+    let serverFunction = firebase.functions().httpsCallable("getEncounterTranscript");
+    serverFunction({userId: user.uid, encounterId: encounterId}).then(function (response) {
+        console.log("Downloaded transcript to memory");
+        console.log(response.data);
+        let newTranscript = response.data.split ('\n').map((item: React.ReactNode, i: string | number | undefined) => <div key={i}>{item}</div>); 
+        setTranscript(newTranscript);
+    }).catch((err) => {
+        console.log(err);
+    });
+  }, [encounters, transcript, setTranscript]);
+
+  React.useEffect(() => {
+    refreshEncounters();
+    let timerHandle = setInterval(refreshEncounters, 10 * 1000);
+    return () => {
+      if (timerHandle) {
+        clearInterval(timerHandle);
+      }
+    };
+  }, [refreshEncounters, refresh]);
+
+  const onView = React.useCallback((event: any, index: number) => {
+    fetchTranscript(encounters[index].encounterId);
+    setViewOpen(true);
+  }, [encounters, setViewOpen, setViewEncounterIndex]);
```

Comment:
shouldn't be dependent on setViewEncounterIndex, or at least, you aren't calling it...